File: demo-liquid-glass/DemoLiquidGlass/domain/models/Course.swift
// domain/models/Course.swift
// Pure Swift struct — NO SwiftUI, NO Foundation imports
// Dependency rule: domain layer has zero external framework dependencies

public struct Course: Identifiable, Equatable {
    public let id: String
    public let name: String
    public let semester: String
    public let enrollmentCode: String
    public let studentCount: Int

    public init(
        id: String,
        name: String,
        semester: String,
        enrollmentCode: String,
        studentCount: Int
    ) {
        self.id = id
        self.name = name
        self.semester = semester
        self.enrollmentCode = enrollmentCode
        self.studentCount = studentCount
    }
}

File: demo-material3/lib/features/peer_evaluation/domain/models/course.dart
// Pure Dart — NO Flutter imports
class Course {
  final String id;
  final String name;
  final String semester;
  final String enrollmentCode;
  final int studentCount;

  const Course({
    required this.id,
    required this.name,
    required this.semester,
    required this.enrollmentCode,
    required this.studentCount,
  });
}
